controllers/dscinitialization/monitoring.go

```diff
@@ -418,7 +418,7 @@ func createMonitoringProxySecret(ctx context.Context, cli client.Client, name st
 	}
 
 	foundProxySecret := &corev1.Secret{}
-	err = cli.Get(ctx, client.ObjectKey{Name: name, Namespace: dsciInit.Spec.Monitoring.Namespace}, foundProxySecret)
+	err = cli.Get(ctx, client.ObjectKeyFromObject(desiredProxySecret), foundProxySecret)
```

Comment:
is this actually correct ? `client.ObjectKeyFromObject` would work only if the object has the name and namespace set, in this case, it seems `foundProxySecret` does not have a name & namespace.

same problem seems to affect the other changes.

Reply:
isnt only need name/namespace set in the desiredProxySecret and the result found obj will be stored in foundProxySecret?

Reply:
oh sorry, got confused by the github web ui, please ignore it

Reply:
i think steps we are using:
- create `designed*`
- do a `Get()` call 
- if not found as IsNotFound, then call `Create(,desired*)`

Reply:
hum, this does feel a bit waste in the case of keep reconcilation.

Reply:
@lburgazzoli I was wondering looking at such code as well, what is the proper sequence? Create() if it is AlreadyExists, Get()? Since Get()-Create() impossible to do atomically

Reply:
i think what really need to do is:
do a Get() by name and namespace
then depends on IsNotFound to do a Create() with desired*
then maybe on the safe side also have a check on IsAlreadyExists() with ownerreference? 

depends on the kind of the resource:
- for namespace, it might not be a problem, if something else changed existing resource.
- but for secret or configmap, we should have Update() as reconcile than Create()

Reply:
Ideally, just to use server side apply and have a single atomic call (no need for get/create/update), if server side is not applicable, maybe use `controllerutil.CreateOrUpdate`. 

I guess the tricky part if for secrets as we generate a key and we should not change it for every reconcile loop.

Reply:
some snippet https://github.com/opendatahub-io/opendatahub-operator/pull/1303 if this can be used ?

Reply:
> Ideally, just to use server side apply and have a single atomic call (no need for get/create/update), if server side is not applicable, maybe use `controllerutil.CreateOrUpdate`.

Looked there. What about CreateOrPatch?

Reply:
that's ok too